addons/web/static/src/webclient/actions/client_actions.js

```diff
@@ -124,3 +124,13 @@ async function softReload(env, action) {
 }
 
 registry.category("actions").add("soft_reload", softReload);
+
+/**
+ * Client action to switch the selected company
+ * Serves as a way to change the company from a backend call
+ */
+async function switchCompany(env, action) {
+    env.services.company.setCompanies([action.params.company], false);
+}
+
+registry.category("actions").add("switch_company", switchCompany);
```

Comment:
mmmh, what about using the `company` service that already exists (see the line `this.companyService.setCompanies(taxUnit.company_ids);` in account_report), and maybe do all of this inside the account module (and extend `install_demo` as well) ?